### ISSUE TITLE
localFile is coming null in the Prismic response.

### ISSUE THREAD

**Issue by HammadUllahKhan12** [prismicio/prismic-gatsby#454]
We recently update the gatsby packages and use the gatsby-image-plugin because gatsby-image is deprecated.
But we are facing this the all the prismic endpoints return the localFile null because of this we were unable to load the image.
I am share some of code snippet. Please have a look & let me know if i miss something. 

This is the code-snippet example :
```
  query MainBlogPage($slug: String) {
    allPrismicBlogPost(sort: { fields: data___publish_date, order: DESC }) {
      edges {
        node {
          uid
          url
          data {
            featured_blog
            navigation_tag {
              tags
            }
            headline {
              text
            }
            hero_image {
              alt
              localFile {
                childImageSharp {
                  fluid {
                    ...GatsbyImageSharpFluid
                  }
                }
              }
            }
          }
        }
      }
      totalCount
    }
```
and here is the screenshot of the GQL playground.
![Screenshot from 2021-09-30 10-53-50](https://user-images.githubusercontent.com/48718737/135479221-ad7b9156-b8ce-4d98-8521-bd20962f57cb.png)

**Comment by angeloashmore** [comment]
Hi @HammadUllahKhan12, which version of `gatsby` and `gatsby-source-prismic` are you using? You can run the following command to list all of your plugins and versions: `npx gatsby info`

Your query looks okay. If it shows `null`, it means either a file was not selected for the field (not the case here), or the plugin did not download the file.

---

By the way, I would recommend using the Imgix URLs rather than `localFile.childImageSharp`. Using Imgix means Gatsby does not have to download and process your images which results in much quicker build times.

You can learn more here: https://prismic.io/docs/technologies/template-images-gatsby#image-processing

**Comment by HammadUllahKhan12** [comment]
hey @angeloashmore Thanks for your quick feedback. I appreciate that.

In our gatsby-config.js we have configured 
```
   shouldDownloadImage: ({ node, key, value }) => {
          // Return true to download the image or false to skip.
          return true
        },
```
Due to some internet connectivity issue. It gives timeout error. Do we have any way to increase the time out for this.

**Comment by angeloashmore** [comment]
@HammadUllahKhan12 I see, this means you are using `gatsby-source-prismic` V3. The plugin will download all images *unless* that function returns `false`. In most cases, the code snippet you posted above is good.

You could try to reduce the number of concurrent downloads by setting the `GATSBY_CONCURRENT_DOWNLOAD` environment variable. By default, that value is set to 200 concurrent downloads.

```sh
GATSBY_CONCURRENT_DOWNLOAD=1 npm run develop
```

This could cause your start-up and build times to increase since downloads will happen slower, so you can adjust the number to what works for you.

**Comment by angeloashmore** [comment]
@tiste This is now supported in `v5.0.1`. See #470 for more details.

Here's a quick example of what you can provide to `shouldDownloadFiles` now:

```js
{
  // Boolean determining if all files should be downloaded.
  shouldDownloadFiles: true,

  // Function returning a boolean determining if all files should or should not be downloaded.
  // It is given the image or link field to download.
  shouldDownloadFiles: (field) => true,

  // Object of specific fields. If a field is not listed here, it is not downloaded
  shouldDownloadFiles: {
    // Boolean determining if the file should be downloaded.
    'myCustomType.data.myImageField': true,

    // Function returning a boolean determining if the file should be downloaded.
    // It is given the image or link field to download.
    'myCustomType.data.myImageField': (field) => true,
  }
}
```

**Comment by tiste** [comment]
@angeloashmore great! Many thanks.

**Comment by tiste** [comment]
I just tested it may interest you @angeloashmore.
From a clean install, it works pretty well, all images are there, and GraphQL queries are filled in.
But if I shut down gatsby and relaunch it, it seems it loses all the image path, and GraphQL queries became `null`.

**Comment by angeloashmore** [comment]
@tiste Thanks for the report. I just published `v5.0.3` with a fix that should that issue.

It turned out to be a garbage collection issue. More details are given here: https://github.com/prismicio/prismic-gatsby/pull/472
